packages/components/src/hooks/useUniqueId.test.ts

```diff
@@ -0,0 +1,34 @@
+import { renderHook } from '@testing-library/react';
+
+import { useUniqueId } from './useUniqueId';
+
+describe('useUniqueId', () => {
+  it('returns a unique ID from each call', () => {
+    // Act
+    const { result: hookReturnRef1 } = renderHook(useUniqueId);
+    const { result: hookReturnRef2 } = renderHook(useUniqueId);
+
+    // Assert
+    expect(hookReturnRef1.current).not.toEqual(hookReturnRef2.current);
+  });
+
+  it('returns the same ID on each render', () => {
+    // Act
+    const { result: hookReturnRef, rerender } = renderHook(useUniqueId);
+    const resultAfterFirstRender = hookReturnRef.current;
+
+    rerender();
+    const resultAfterSecondRender = hookReturnRef.current;
+
+    // Assert
+    expect(resultAfterSecondRender).toEqual(resultAfterFirstRender);
+  });
+
+  it('returns a string that starts with the given prefix', () => {
+    // Act
+    const { result: hookReturnRef } = renderHook(useUniqueId, { initialProps: 'test-prefix' });
+
+    // Assert
```

Comment:
I'm still not convinced these comments pay for themselves, but, more generally, this is a really nice test.

Reply:
Especially in simple tests, I agree that they don't have much descriptive value for a reader.

I find their value to be in writing tests. Forming the habit of always writing these comments serves as a useful reminder on how to structure the test.

Reply:
yup, habit and test structure is a key motivation.